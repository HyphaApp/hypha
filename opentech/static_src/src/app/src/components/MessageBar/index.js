import React from 'react'
import PropTypes from 'prop-types'

import { MESSAGE_TYPES } from '@actions/messages'

const MessageBar = ({message, onDismiss}) => {

    return (
        <div className={message.type}>
            <p>{message.message}</p>
            {onDismiss && <button onClick={onDismiss}>[X]</button>}
        </div>
    )
}

MessageBar.propTypes = {
    message: PropTypes.shape({
        type: PropTypes.oneOf(Object.values(MESSAGE_TYPES)),
        message: PropTypes.string
    }),
    onDismiss: PropTypes.func,
}

export default MessageBar
