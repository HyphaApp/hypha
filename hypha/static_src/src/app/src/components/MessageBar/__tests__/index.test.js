import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import MessageBar from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test inline loading component", () => {
    const message = "message text"
    const dismissMessage = "dismiss message"
    const onDismiss = jest.fn()
    const type = 'success'

    const subject = mount(<MessageBar
        message={message}
        dismissMessage={dismissMessage}
        onDismiss={onDismiss}
        type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.messages__text--success').length).toBe(1);
      subject.find('button').props().onClick()
      expect(onDismiss).toHaveBeenCalled();
      expect(subject.find('button').text()).toBe(dismissMessage)
    });
  
    test('render a inline loading component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
