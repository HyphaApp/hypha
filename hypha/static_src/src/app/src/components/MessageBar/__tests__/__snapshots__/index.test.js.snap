// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test inline loading component render a inline loading component 1`] = `
<MessageBar
  dismissMessage="dismiss message"
  message="message text"
  onDismiss={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  type="success"
>
  <li
    className="messages__text messages__text--success"
  >
    <div
      className="messages__inner"
    >
      <p
        className="messages__copy"
      >
        message text
      </p>
      <button
        className="button messages__button"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
      >
        dismiss message
      </button>
    </div>
  </li>
</MessageBar>
`;
